addons/base_import/tests/test_base_import.py

```diff
@@ -468,6 +468,37 @@ def test_parse_relational_fields(self):
         self.assertEqual(float(result[0][-1]), 5.69)
         self.assertEqual(str(result[0][-2]), '2017-10-12')
 
+    def test_parse_scientific_notation(self):
+        """ Ensure that scientific notation is correctly converted to decimal """
+        import_wizard = self.env['base_import.import']
+
+        test_data = [
+            {
+                "input":              "1E+05",
+                "expected_result":    "100000.000000",
+            },
+            {
+                "input":              "1.20E-05",
+                "expected_result":    "0.000012",
+            },
+            {
+                "input":              "1,9e5",
+                "expected_result":    "190000.000000",
+            },
+            {
+                "input":              "9,5e-5",
+                "expected_result":    "0.000095",
+            },
+            {
+                "input":              "not a sci notation",
+                "expected_result":    "not a sci notation",
+            },
+        ]
+
+        for x in test_data:
+            result = import_wizard._convert_scientific_notation_to_float(x["input"])
```

Comment:
You should test the _parse_float_from_data complete behaviour, not only the new function.